File: Python/Q14_Singleton.py
class Configuracao(object):
  _instances = {}
  def __new__(class_, *args, **kwargs):
    if class_ not in class_._instances:
        class_._instances[class_] = super(Configuracao, class_).__new__(class_, *args, **kwargs)
    return class_._instances[class_]

config = Configuracao()
config2 = Configuracao()
print(config == config2)

File: go/Q14_Singleton/Q14_Singleton.go
package main

import (
	"fmt"
	"sync"
)

type Configuracao struct {
}

var instancia *Configuracao
var once sync.Once

func GetConfiguracao() *Configuracao {
	once.Do(func() {
		instancia = &Configuracao{}
	})
	return instancia
}

func main() {
	config := GetConfiguracao()
	config2 := GetConfiguracao()
	fmt.Println( config == config2)
}
